addons/account/models/account.py

```diff
@@ -253,6 +256,24 @@ def _check_currency(self):
             if self.default_debit_account_id and not self.default_debit_account_id.currency_id.id == self.currency_id.id:
                 raise UserError(_('Configuration error!\nThe currency of the journal should be the same than the default debit account.'))
 
+    @api.one
```

Comment:
> The only optional one is to check that bank_account_id.partner_id ==
> self.company_id.partner_id
> It is arguable whether it makes sense or not to set the partner_id of a
> journal's bank account since we know it belongs to the company.
> But since the field is there, and using bank_account_id.partner_id =
> self.company_id.partner_id was the old way of finding that a bank
> account belongs to the company, I figured out it is safer to enforce
> that value for this field.

Not sure it's a good idea. You may end in deadlocks since the partner
field is not even visible if the form view. I would remove this
constraint if possible, it is too complex for the end-user to fix this
if the exception is triggered.

Having a holding that manages bank accounts of different companies is a
use case.


-- 
Fabien Pinckaers
Odoo Founder

Web: https://www.odoo.com
Twitter: @fpodoo

Instant Demo: https://odoo.com/start/cms